dask/dataframe/io/orc/core.py

```diff
@@ -49,6 +49,7 @@ def __call__(self, parts):
         )
         if self.index:
             _df.set_index(self.index, inplace=True)
```

Comment:
Just an informative question, why do we keep the `inplace=True` in here?

Reply:
I'm keeping `inplace=True` in non-testing code to try to minimize data copies. To be clear, I'm not 100% sure whether or not copies would still be avoided if `inplace=False` (maybe @jorisvandenbossche or @mroeschke have more insight here)

Reply:
Btw this is definitely a fair question to ask. In general, I think avoiding inplace operations make things easier to reason about. Let's handle as needed in a follow-up PR

Reply:
Looking at `set_index` in pandas, a copy of the entire DataFrame is avoided if `inplace=True`

(Part of the motivation of the original deprecation is that a copy can always be avoid since the index is only being modified)